src/shared/conf-parser.c

```diff
@@ -910,6 +910,38 @@ int config_parse_safe_string(
         return free_and_strdup_warn(s, empty_to_null(rvalue));
 }
 
+int config_parse_safe_ascii_string(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        char **s = data;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (!string_is_safe(rvalue)) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Specified string contains unsafe characters, ignoring: %s", rvalue);
```

Comment:
This print statement will not work well. If the string has control chars, the log will end up garbled. Either journalctl will refuse to show it w/o `--all`, or even the message could be truncated at the first newline. The argument needs to be passed through cescape() before printing, or not printed at all. I think it's worth the trouble here to print it, because the user most likely inserted some ccs by mistake in the config and a good error message will be useful to them.